### PR TITLE
refactor: Consolidate creator, storage and resolver DID under VDRI

### LINKED ISSUE #777: Consolidate creator, storage and resolver DID under VDRI
Consolidate creator, storage and resolver under vdr

### PR DESCRIPTION
**Pull request by fqutishat**
closes #777

Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

**Comment by fqutishat** [comment]
@sandrask @talwinder50 @sudeshrshetty

File: cmd/aries-agentd/startcmd/start.go [hunk 1]
```diff
@@ -301,7 +282,7 @@ func startAgent(parameters *agentParameters) error {
 		opts = append(opts, defaults.WithStorePath(parameters.dbPath))
 	}
 
-	resolverOpts, err := getResolverOpts(parameters.defaultLabel, parameters.dbPath, parameters.httpResolvers)
+	resolverOpts, err := getResolverOpts(parameters.httpResolvers)
```
Comment (sudeshrshetty) [inline cmd/aries-agentd/startcmd/start.go]: nice, more cleaner now.

File: pkg/vdri/peer/vdri.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
+/*
```
Comment (rolsonquadras) [inline pkg/vdri/peer/vdri.go]: looks like unit test is missing for this file. ie `vdri_test.go`

Comment (fqutishat) [reply pkg/vdri/peer/vdri.go]: it's already test it with other test file in the same package

File: pkg/didcomm/protocol/didexchange/states.go [hunk 3]
```diff
@@ -400,12 +400,12 @@ func (ctx *context) getDIDDocAndConnection(pubDID string) (*did.Doc, *Connection
 	logger.Debugf("creating new '%s' did for connection", didMethod)
 
 	// by default use peer did
-	newDidDoc, err := ctx.didCreator.Create(didMethod)
+	newDidDoc, err := ctx.vdriRegistry.Create(didMethod)
```
[the remark this answers is not in the document]
Comment (fqutishat) [reply pkg/didcomm/protocol/didexchange/states.go]: done

File: pkg/framework/aries/api/vdri/vdri.go [hunk 4]
```diff
@@ -0,0 +1,124 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package didcreator
```
[the remark this answers is not in the document]
Comment (fqutishat) [reply pkg/framework/aries/api/vdri/vdri.go]: done

File: pkg/framework/aries/api/vdri/vdri.go [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+	"github.com/hyperledger/aries-framework-go/pkg/doc/did"
+)
+
+// ErrNotFound is returned when a DID resolver does not find the DID.
+var ErrNotFound = errors.New("DID not found")
+
+// Registry vdri registry
+type Registry interface {
+	Resolve(did string, opts ...ResolveOpts) (*did.Doc, error)
+	Put(doc *did.Doc) error
```
[the remark this answers is not in the document]
Comment (fqutishat) [reply pkg/framework/aries/api/vdri/vdri.go]: done

File: pkg/didcomm/protocol/didexchange/service_test.go [hunk 6]
```diff
@@ -530,7 +527,7 @@ type mockStore struct {
 	get func(string) ([]byte, error)
 }
 
-// Put stores the key and the record
+// Store stores the key and the record
```
Comment (troyronda) [inline pkg/didcomm/protocol/didexchange/service_test.go]: The comment was updated but not the method name?

File: test/bdd/pkg/didexchange/didexchange_sdk_steps.go [hunk 7]
```diff
@@ -18,7 +18,7 @@ import (
 	"github.com/hyperledger/aries-framework-go/pkg/client/didexchange"
 	"github.com/hyperledger/aries-framework-go/pkg/didcomm/common/service"
 	diddoc "github.com/hyperledger/aries-framework-go/pkg/doc/did"
-	"github.com/hyperledger/aries-framework-go/pkg/framework/didresolver"
+	vdriapi "github.com/hyperledger/aries-framework-go/pkg/framework/aries/api/vdri"
```
Comment (troyronda) [inline test/bdd/pkg/didexchange/didexchange_sdk_steps.go]: TBD: best location to have these framework SPI folders.

File: pkg/didcomm/protocol/didexchange/service.go [hunk 8]
```diff
@@ -60,8 +58,7 @@ type provider interface {
 	StorageProvider() storage.Provider
 	TransientStorageProvider() storage.Provider
 	Signer() kms.Signer
-	DIDResolver() didresolver.Resolver
-	DIDStore() didstore.Storage
+	VDRIRegistry() vdriapi.Registry
```
Comment (troyronda) [inline pkg/didcomm/protocol/didexchange/service.go]: Technically this has some stutter as “R” stands for registry.

File: pkg/framework/aries/api/vdri/vdri.go [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+
+// Registry vdri registry
+type Registry interface {
+	Resolve(did string, opts ...ResolveOpts) (*did.Doc, error)
+	Store(doc *did.Doc) error
+	Create(method string, opts ...DocOpts) (*did.Doc, error)
+}
+
+// VDRI verifiable data registry interface
+type VDRI interface {
```
Comment (sudeshrshetty) [inline pkg/framework/aries/api/vdri/vdri.go]: Question: Should VDRI implement closer interface? to release/dispose underlying store or other resources?

File: pkg/framework/aries/api/vdri/vdri.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (fqutishat) [reply pkg/framework/aries/api/vdri/vdri.go]: fixed https://github.com/hyperledger/aries-framework-go/pull/789#discussion_r346480686

File: pkg/storage/leveldb/leveldb_store.go [hunk 10]
```diff
@@ -112,7 +112,7 @@ type leveldbStore struct {
 	db *leveldb.DB
 }
 
-// Put stores the key and the record
+// Store stores the key and the record
```
Comment (troyronda) [inline pkg/storage/leveldb/leveldb_store.go]: Again - comment updated but not method? In a few places actually.

File: pkg/framework/aries/api/vdri/vdri.go (hunk 9 above)
Comment (fqutishat) [reply pkg/framework/aries/api/vdri/vdri.go]: Yes I will implement it in separate PR

File: pkg/storage/leveldb/leveldb_store.go (hunk 10 above)
Comment (sudeshrshetty) [inline pkg/storage/leveldb/leveldb_store.go]: wrong method name

File: pkg/storage/mem/mem_store.go [hunk 11]
```diff
@@ -95,7 +95,7 @@ type memStore struct {
 	sync.RWMutex
 }
 
-// Put stores the key and the record
+// Store stores the key and the record
```
Comment (sudeshrshetty) [inline pkg/storage/mem/mem_store.go]: wrong method name

File: pkg/storage/store.go [hunk 12]
```diff
@@ -25,7 +25,7 @@ type Provider interface {
 
 // Store is the storage interface
 type Store interface {
-	// Put stores the key and the record
+	// Store stores the key and the record
```
Comment (sudeshrshetty) [inline pkg/storage/store.go]: wrong method name

Comment (fqutishat) [reply pkg/storage/store.go]: done

File: pkg/storage/mem/mem_store.go (hunk 11 above)
Comment (fqutishat) [reply pkg/storage/mem/mem_store.go]: done

File: pkg/storage/leveldb/leveldb_store.go (hunk 10 above)
Comment (fqutishat) [reply pkg/storage/leveldb/leveldb_store.go]: done

Comment (fqutishat) [reply pkg/storage/leveldb/leveldb_store.go]: done

File: pkg/didcomm/protocol/didexchange/service_test.go (hunk 6 above)
Comment (fqutishat) [reply pkg/didcomm/protocol/didexchange/service_test.go]: done

**Pull request merged**